Source/FieldSolver/FiniteDifferenceSolver/HybridPICModel/HybridPICModel.H

```diff
@@ -184,7 +190,23 @@ public:
     std::string m_Jz_ext_grid_function = "0.0";
     std::array< std::unique_ptr<amrex::Parser>, 3> m_J_external_parser;
     std::array< amrex::ParserExecutor<4>, 3> m_J_external;
-    bool m_external_field_has_time_dependence = false;
+    bool m_external_jfield_has_time_dependence = false;
+
+    /** External magnetic field */
+    std::string m_Bx_ext_grid_function = "0.0";
+    std::string m_By_ext_grid_function = "0.0";
+    std::string m_Bz_ext_grid_function = "0.0";
+    std::array< std::unique_ptr<amrex::Parser>, 3> m_B_external_parser;
+    std::array< amrex::ParserExecutor<4>, 3> m_B_external;
+    bool m_external_bfield_has_time_dependence = false;
+
+    // Path of the file where external
+    // fields are stored
+    std::string m_external_j_fields_path = "0.0";
+    std::string m_external_b_fields_path = "0.0";
+
+    std::string m_J_ext_grid_s = "0.0";
+    std::string m_B_ext_grid_s = "0.0";
```

Comment:
Being a bit more verbose here would help to make the code more readable.
```suggestion
    std::string m_J_ext_grid_style = "0.0";
    std::string m_B_ext_grid_style = "0.0";
```